waspc/data/Generator/templates/server/src/auth/providers/email/signup.ts

```diff
@@ -0,0 +1,31 @@
+import waspServerConfig from '../../../config.js';
+import { EmailSender, EmailFromField } from "../../../email/core/types.js";
+import { handleRejection } from "../../../utils.js";
+import { createEmailVerificationToken, createUser } from "../../utils.js";
+import { GetVerificationEmailContentFn } from './types.js';
+
+export function getSignupRoute({
+    emailSender,
+    fromField,
+    getVerificationEmailContent,
+}: {
+    emailSender: EmailSender;
+    fromField: EmailFromField;
+    getVerificationEmailContent: GetVerificationEmailContentFn;
+}) {
+    return handleRejection(async (req, res) => {
+        const userFields = req.body || {};
+    
+        const user = await createUser(userFields);
+    
+        const verifyEmailToken = await createEmailVerificationToken(user);
+        const verificationLink = `${waspServerConfig.backendUrl}/auth/email/verify-email?token=${verifyEmailToken}`;
```

Comment:
We need to now know about the backend URL as well and the user needs to provide it through the env variables.

Reply:
The users won't see this URL at all since it will redirect to the client URL right after it processes the token it received.

In the RFC we had two options:
1. Load the client page first 🔵
    1. Client page loads with `?token`
    2. It sends a backend request with the token
    3. User is verified
2. Go directly to the backend 🟠
    1. Backend loads with `?token`
    2. Backend redirects to a success or failure page
    
  The second 🟠 one seemed simpler at the time. But now that I implemented it, I see that we need to introduce a mandatory env variable called `BACKEND_URL` to get it working properly. Maybe the first 🔵 one is better for the user after all?